lib/client-options.js

```diff
@@ -129,6 +129,61 @@ function extend(baseOptions, userOptions) {
 }
 
 /**
+ * Creates a new instance of query options with the values from the user.
+ * When some values are not defined, it takes the default values from
+ * - {@link ExecutionProfile}.
+ * - {@link QueryOptions} from the default options.
+ * @param {Client} client
+ * @param {QueryOptions|function} userOptions
+ * @param {Function} [rowCallback]
+ * @param {Boolean} [logged]
+ */
+function createQueryOptions(client, userOptions, rowCallback, logged) {
+  var profile =
+    client.profileManager.getProfile(userOptions && userOptions.executionProfile) || utils.emptyObject;
```

Comment:
I really, really like how this is the only place the profile is resolved, and then is attached to created options, it keeps the rest of the code very simple.  There isn't a lot of duplicated logic all over the place around 'is it on the query options?  is it on the profile? is it on the base options?' and so on, it's all contained here 👍